### PR TITLE
cli: Add var args to plan output.

### PR DESCRIPTION
**Pull request by Liraim**
Fixes for issue #11616.

### CODE REVIEW

File: command/job_plan.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 const (
 	jobModifyIndexHelp = `To submit the job with version verification run:
 
-nomad job run -check-index %d %s
+nomad job run -check-index %d %s %s
```
Comment (tgross) [inline command/job_plan.go:18]: I've checked three scenarios here...

<details><summary>(1) Job with a variable that's set, LGTM</summary>

```
$ nomad job plan -var message="hello, world" ./example.nomad
+ Job: "httpd"
+ Task Group: "web" (1 create)
  + Task: "http" (forces create)

Scheduler dry-run:
- All tasks successfully allocated.

Job Modify Index: 0
To submit the job with version verification run:

nomad job run -check-index 0 -var="message=hello, world"  ./example.nomad

When running the job with the check-index flag, the job will only be run if the
job modify index given matches the server-side version. If the index has
changed, another user has modified the job and the plan's results are
potentially invalid.
```

</details>

<details><summary>(2) Job with a variable that's not set (just to make sure we don't panic on the error message), LGTM</summary>

```
$ nomad job plan ./example.nomad
Error getting job struct: Error parsing job file from ./example.nomad:
<nil>: Unset variable "message"; A used variable must be set or have a default value; see https://www.nomadproject.io/docs/job-specification/hcl2/variables for details.
```

</details>

(3) Job without a variable:

```
$ nomad job plan ./example.nomad
+ Job: "httpd"
+ Task Group: "web" (1 create)
  + Task: "http" (forces create)

Scheduler dry-run:
- All tasks successfully allocated.

Job Modify Index: 0
To submit the job with version verification run:

nomad job run -check-index 0  ./example.nomad

When running the job with the check-index flag, the job will only be run if the
job modify index given matches the server-side version. If the index has
changed, another user has modified the job and the plan's results are
potentially invalid.
```

On that last scenario, we get an ugly extra space between the job name and the empty string for the variable. I think that'll be removed and look correct if we remove the extra space we added here?

```suggestion
nomad job run -check-index %d %s%s
```

Comment (Liraim) [reply command/job_plan.go]: Thanks for pointing this, in fact this ugly space present in first example too but your fix is removing that.

Comment (tgross) [reply command/job_plan.go]: 🤦 I totally missed that. That's why we review each other! 😀

**Review by tgross — APPROVED** [review]
LGTM! I've added a changelog entry and once CI is done this will get merged. Thanks again!

**Pull request merged**